### ISSUE TITLE
Vendored static frameworks header search paths missing in child target

### ISSUE THREAD

**Issue by benasher44** [CocoaPods/CocoaPods#5173]
- [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).
# Report
## What did you do?

Built a project that uses CocoaPods that defines vendored static frameworks in its main target, and the main target's child inherits its search paths.
## What did you expected to happen?

I expected the project to build.
## What happened instead?

The build failed claiming that it could not find some headers (in a `#import` statement) for the imported headers in the child target
## Podfile

``` ruby
platform :ios, '8.0'
use_frameworks!

source 'https://github.com/CocoaPods/Specs.git'
source 'https://github.com/ooyala/ios-sdk-podspecs.git'

abstract_target 'SharedDeps' do
  pod 'AFNetworking'
  pod 'OoyalaSDK'

  target 'InheritSearchPathsBroken' do
    abstract_target 'TestsShared' do
      inherit! :search_paths
      pod 'OCMock'
      target 'InheritSearchPathsBrokenTests'
    end
  end

  target 'InheritSearchPathsBroken WatchKit 1 Extension'
end
```

I re-used a sample project from an [earlier ticket](/CocoaPods/CocoaPods/issues/4943), and I've [updated this project](/benasher44/CocoaPods4943) for this ticket and for 1.0.0.beta.8.

**Comment by neonichu** [comment]
Just linking [this comment](https://github.com/CocoaPods/CocoaPods/issues/5221#issuecomment-215537506) here which adds some background on the issue and a potential fix.

**Comment by fcy** [comment]
@neonichu answering this comment. If you go to the sample code in #5303 and add to `MissingPodTests`'s Framework Search Path the same path that is in `MissingPod` target (aka `"${PODS_ROOT}/Headers/Public/Crashlytics"`) it works. I didn't quite get the "deep path" thing.

Also the workaround mentioned of duplicating the dependency don't actually fully work since it makes mocking stubbing impossible because the classes in the test target are not the same as the ones in the app.

So I guess my question is, why adding `"${PODS_ROOT}/Headers/Public/Crashlytics"` to the test target not a valid solution?

**Comment by fcy** [comment]
I'll try to summarize my point of view in this issue better.

1. Given the same set of dependencies (no vendored binary ones) in a Podfile it should work the same with and without `use_frameworks!`. Meaning CocoaPods should resolve all the dependencies to make everything build.
2. If we want to use Dynamic Frameworks ability to better control what dependencies are propagate and which aren't we should add extra settings for this special case. Something like `s.dependency 'ABC', private: true` this new setting wouldn't do anything when using static libraries but would better organize the dependencies of dependencies when using Dynamic Frameworks

**Comment by coolnalu** [comment]
saw this from XCode 7.3.1 release note, any new information here?

> Swift
>
> The Swift compiler is now stricter about including non-modular header files. Debugging a Swift target requires that frameworks be properly modular, meaning all of the publicly-imported headers are either accounted for in the framework's umbrella header, imported from another modular framework, or listed explicitly in a custom module.modulemap file (advanced users only).
The compiler will run into issues if the same header file is accessible both through Header Search Paths (-I, -isystem) and Framework Search Paths (-F, -iframework), even if there are symbolic links involved. In these cases, you should prefer using Framework Search Paths alone. (23341162)
>
> Note: This invalid configuration may be generated by external systems, such as CocoaPods.

**Comment by neonichu** [comment]
The second copy of headers in `PODS_ROOT/Headers` shouldn't actually exist anymore, have to investigate why it still does.

**Comment by fcy** [comment]
What?

**Comment by fcy** [comment]
I'd have elaborated more but I really didn't understand it.

**Comment by neonichu** [comment]
I was talking about this change #5146

**Comment by fcy** [comment]
Humm 😕. That change talks about header search path, I was talking about framework search path. Not sure we are on the same page. I've reworded [my question|https://github.com/CocoaPods/CocoaPods/issues/5173#issuecomment-218924439] to try to make clearer:

Why adding `"${PODS_ROOT}/Headers/Public/Crashlytics"` that is already present in the app target Framework Search Path to the test target Framework Search Path is not a valid solution?

**Comment by neonichu** [comment]
I assumed you meant header search path, because adding a directory of headers to the framework search path accomplishes nothing.

**Comment by fcy** [comment]
Oh I see, my bad I double clicked the wrong entry in Build Settings. Thanks for explaining.

**Comment by benasher44** [comment]
This should be fixed as part of #5514.
